src/Security/Member.php

```diff
@@ -1696,6 +1696,13 @@ public function changePassword($password)
             $this->write();
         }
 
+        $resultFromExtensions = $this->extend('onAfterChangePassword', $password, $valid);
```

Comment:
If you modify the result from valid to invalid here, you have still written the member to disk in a supposedly invalid state.

I suggest 'onBeforeChangePassword' (takes $valid result and can modify it before $valid->isValid() is called), and an `onAfterChangePassword` which can take the result but not modify it.

Your extensions should instead be responsible for doing combineAnd() not this code. This seems like baked-in logic for your specific use case.